backend/capellacollab/sessions/hooks/jupyter.py

```diff
@@ -34,7 +40,10 @@ def __init__(self):
 
     def configuration_hook(
         self,
+        db: orm.Session,
         user: users_models.DatabaseUser,
+        tool: tools_models.DatabaseTool,
+        token: dict[str, t.Any],
         **kwargs,
     ) -> tuple[
         JupyterConfigEnvironment,
```

Comment:
> I think the `JupyterConfigEnvironment` is missing the `JUPYTER_URI` field.

Is added.

> In addition the `configuration_hook` shows a mypy error for me (as `JupyterConfigEnvironment` is not considered to be a `dict[str, str]` even though it is. Do you know any good way to handle this?

A simple `environment: JupyterConfigEnvironment` seems to do it.

> Furthermore, the expected arguments of the `HookRegistration` and the `JupyterIntegration` are slightly different, e.g., the operator is missing. Isn't this somehow required or should be done when using inheritance here?

This is possible because we always call the hooks with named arguments, never with positional arguments. The intention is to remove the number of unused arguments on the different integrations. Each integration should just "request" the parameters, which are required.